packages/ui/src/ui/pages/query-tracker/QueryTrackerTopRow/QueryNameForm.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {ChangeEvent, FC, KeyboardEvent, useCallback, useRef, useState} from 'react';
+import {Button, Icon, Popup, Text, TextInput, Tooltip} from '@gravity-ui/uikit';
+import PencilIcon from '@gravity-ui/icons/svgs/pencil.svg';
+import XmarkIcon from '@gravity-ui/icons/svgs/xmark.svg';
+import CheckIcon from '@gravity-ui/icons/svgs/check.svg';
+import cn from 'bem-cn-lite';
+import './QueryNameForm.scss';
+
+const ELEMENT_SIZE = 'l';
+const NAME_PLACEHOLDER = 'No name';
+
+const form = cn('query-name-form');
+const popup = cn('query-name-popup');
+
+type Props = {
+    queryName?: string;
+    onSave: (name: string | undefined) => void;
+};
+
+export const QueryNameForm: FC<Props> = ({queryName, onSave}) => {
```

Comment:
I hope we will replace the component with inline editing, let's wait for an answer for:
https://nda.ya.ru/t/PqZgJWeF74wEw5

Additionally we have `EditableAsText` component it might be useful to replace some parts of current implementation.